Add additional documentation to application spec (#631)
cdr/docs: workspaces/applications.md

## Application specification file

> In order to specify custom applications, add a configuration file at path

Reviewer: ```suggestion
To define custom applications, add a configuration file at
```

    command: /home/coder/.local/bin/projector
    # Array of arguments for command. Optional.
    args: ["run"]
    # Health checks to get running application status. Can use exec or http
	# health checks to localhost. It is recommended to specify a health-check
	# although not strictly required. If one is not supplied then an http
	# request is sent to the application root path "/".
    health-check:
	  # Exec commands require an exit code of '0' to report healthy.
      exec:
        command: "pgrep"
        args: ["projector"]
	  # http sends a GET request to the address specified via the parameters.
	  # It expects a 200 status code to report healthy.
      http:
	    # Scheme must be "http" or "https". If not specified it inherits
	    # the application scheme. Optional.
        scheme: "http"
		# The host to use when dialing the address. If not specified it
		# inherits the application host. Optional.
		host: "localhost"
		# Port to use when dialing the application. If not specified it
		# inherits the application port. Optional.
        port: 9999
		# Path to use for the health check. If not specified defaults to
		# "/".
        path: "/healthz"
```

**Notes**:

> - A health check _must_ report healthy in order to access the application.

Reviewer: ```suggestion
- A health check _must_ report healthy for you to access the application.
```

> - If both HTTP and Exec health checks are specified then HTTP is preferred.

Reviewer: ```suggestion
- If you specify both the HTTP and Exec health checks, Coder prioritizes HTTP.
```
